src/gromacs/nblib/molecules.h

```diff
@@ -118,39 +121,55 @@ class Molecule
     // TODO: add exclusions based on the unique ID given to the particle of the molecule
     void addExclusion(int particleIndex, int particleIndexToExclude);
 
-    // Specify an exclusion with particle and residue names that have been added to molecule
+    //! Specify an exclusion with particle and residue names that have been added to molecule
     void addExclusion(std::tuple<std::string, std::string> particle,
                       std::tuple<std::string, std::string> particleToExclude);
 
-    // Specify an exclusion with particle names that have been added to molecule
+    //! Specify an exclusion with particle names that have been added to molecule
     void addExclusion(const std::string& particleName, const std::string& particleNameToExclude);
 
     //! add various types of interactions to the molecule
     //! Note: adding an interaction type not listed in InteractionTuple in this class results in a compilation error
     template<class Interaction>
-    void addInteraction(ParticleName particleNameI, ParticleName particleNameJ, Interaction interaction)
+    void addInteraction(const ParticleName& particleNameI,
+                        const ResidueName&  residueNameI,
+                        const ParticleName& particleNameJ,
+                        const ResidueName&  residueNameJ,
+                        Interaction         interaction)
     {
         auto& interactionContainer = pickType<Interaction>(interactionData_);
-        interactionContainer.interactions_.emplace_back(particleNameI, particleNameJ, interaction.name());
-        if (interactionContainer.interactionTypes_.count(interaction.name()) == 0)
-        {
-            interactionContainer.interactionTypes_[interaction.name()] = std::move(interaction);
-        }
+        interactionContainer.interactions_.emplace_back(particleNameI, residueNameI, particleNameJ,
```

Comment:
Can this implementation be moved to the source file?

Reply:
done

Reply:
done